cmd/update-rules/main.go

```diff
@@ -141,15 +145,25 @@ func UpdateRules(rules *config.RepositoryRules, branch, goVer string) {
 			continue
 		}
 
-		// update the rules for branch and its dependencies
-		updateBranchRules(&newBranchRule, branch, goVer)
-
 		var branchRuleExists bool
 		// if the target branch rules already exists, update it
+		// update the rules for branch and its dependencies
+		if !removeRules {
```

Comment:
Can we move the deleteRule codeblock to the initial section of the loop. So other checks need not be performed. We can just delete the rule and continue the loop. WDYT